src/chess.ts

```diff
@@ -766,11 +777,34 @@ export class Chess {
       this._kings[piece.color] = EMPTY
     }
 
+    this._updateCastlingRights()
+
     this._updateSetup(this.fen())
 
     return piece
   }
 
+  _updateCastlingRights() {
```

Comment:
Can you use the piece consts in place if the `'r'` and `'k'` literals here? Thanks!